qt/python/mantidqt/widgets/sliceviewer/presenter.py

```diff
@@ -346,6 +349,45 @@ def overlay_peaks_workspaces(self):
         else:
             self.view.peaks_view.hide()
 
+    def replace_workspace(self, workspace_name, workspace):
```

Comment:
Just to confirm, can this method be called also directly? i.e. if one needs to replace the workspace not through the ADS.

Reply:
Yes it can, but if the new workspace is not mostly the same then sliceviewer will close (eg if it is a different type)